src/java.base/share/classes/java/lang/invoke/VarHandleSegmentViewBase.java

```diff
@@ -25,31 +25,47 @@
 
 package java.lang.invoke;
 
+import java.lang.foreign.MemoryLayout;
+import java.lang.foreign.MemorySegment;
+
 /**
  * Base class for memory segment var handle view implementations.
  */
-abstract sealed class VarHandleSegmentViewBase extends VarHandle permits
-        VarHandleSegmentAsBytes,
-        VarHandleSegmentAsChars,
-        VarHandleSegmentAsDoubles,
-        VarHandleSegmentAsFloats,
-        VarHandleSegmentAsInts,
-        VarHandleSegmentAsLongs,
-        VarHandleSegmentAsShorts {
+final class VarHandleSegmentViewBase extends VarHandle {
```

Comment:
Should the name of this class change so that we drop `Base` ? It's no longer extended by anything.